docs/reference/query-rules/apis/list-query-rulesets.asciidoc

```diff
@@ -35,56 +35,162 @@ Requires the `manage_search_query_rules` privilege.
 
 The following example lists all configured query rulesets:
 
+////
+
+[source,console]
+--------------------------------------------------
+PUT _query_rules/ruleset-1
+{
+    "rules": [
+        {
+            "rule_id": "rule-1",
+            "type": "pinned",
+            "criteria": [
+                {
+                    "type": "exact",
+                    "metadata": "query_string",
+                    "values": [ "puggles" ]
+                }
+            ],
+            "actions": {
+                "ids": ["id1"]
+            }
+        }
+    ]
+}
+
+PUT _query_rules/ruleset-2
+{
+    "rules": [
+        {
+            "rule_id": "rule-1",
+            "type": "pinned",
+            "criteria": [
+                {
+                    "type": "exact",
+                    "metadata": "query_string",
+                    "values": [ "puggles" ]
+                }
+            ],
+            "actions": {
+                "ids": ["id1"]
+            }
+        },
+        {
+            "rule_id": "rule-2",
+            "type": "pinned",
+            "criteria": [
+                {
+                    "type": "fuzzy",
+                    "metadata": "query_string",
+                    "values": [ "pugs" ]
+                }
+            ],
+            "actions": {
+                "ids": ["id2"]
+            }
+        }
+    ]
+}
+
+PUT _query_rules/ruleset-3
+{
+    "rules": [
+        {
+            "rule_id": "rule-1",
+            "type": "pinned",
+            "criteria": [
+                {
+                    "type": "exact",
+                    "metadata": "query_string",
+                    "values": [ "puggles" ]
+                }
+            ],
+            "actions": {
+                "ids": ["id1"]
+            }
+        },
+        {
+            "rule_id": "rule-2",
+            "type": "pinned",
+            "criteria": [
+                {
+                    "type": "fuzzy",
+                    "metadata": "query_string",
+                    "values": [ "pugs" ]
+                }
+            ],
+            "actions": {
+                "ids": ["id2"]
+            }
+        },
+        {
+            "rule_id": "rule-3",
+            "type": "pinned",
+            "criteria": [
+                {
+                    "type": "fuzzy",
+                    "metadata": "query_string",
+                    "values": [ "beagles" ]
+                }
+            ],
+            "actions": {
+                "ids": ["id2"]
+            }
+        }
+    ]
+}
+--------------------------------------------------
+// TESTSETUP
```

Comment:
I don't think query rules are automatically cleared between tests, so a TEARDOWN that does that here would be nice. Or maybe add a step that clears all query rules between tests in https://github.com/elastic/elasticsearch/blob/main/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java#L672?